data/render/src/lib.rs

```diff
@@ -21,6 +23,65 @@ pub fn validate(tags: &Tags, entries: &Vec<Entry>) -> Result<(), Box<dyn Error>>
     Ok(())
 }
 
+#[tokio::main]
+pub async fn check_deprecated(token: std::string::String, entries: &mut Vec<Entry>) -> Result<Vec<Entry>, Box<dyn Error>> {
+    let github = Github::new(
+        String::from("user-agent-name"),
+        Credentials::Token(token),
+    )?;
+
+    let mut entries_tmp: Vec<Entry> = entries.to_vec();
+    for entry in &mut entries_tmp {
+        if entry.source.is_none() {
+            continue;
+        }
+
+        let mut source: &str = entry.source.as_ref().unwrap();
+        if source.chars().last().unwrap() == '/' {
+            source = source.trim_end_matches('/');
+        }
+
+        let components: Vec<&str> = source.split("/").collect();
+        if !(components.contains(&"github.com") && components.len() == 5) {
+            // valid github source must have 5 elements - anything longer and they are probably a
+            // reference to a path inside a repo, rather than a repo itself.
+            continue;
+        }
+
+        let owner = components[3];
+        let repo  = components[4];
+
+        let commit_list = github
+            .repo(owner, repo)
+            .commits()
+            .list()
+            .await;
+
+        let commit_list = match commit_list {
+            Ok(commit_list) => commit_list,
+            Err(_error) => Vec::new(),
+        };
```

Comment:
I was originally using `await?`, but if some repo failed to get commit list, it will only gives me this error message
```
Error: Codec(Error("invalid type: null, expected struct User", line: 1, column: 67983))
```

and the program will directly exit, so thats the reason why I use that method to handle this error.

Not sure if there is any other way to handle this?

Reply:
Does this make more sense?
```rust
        if let Ok(commit_list) = github
            .repo(owner, repo)
            .commits()
            .list()
            .await
        {
            let date = &commit_list[0].commit.author.date;
            let last_commit = NaiveDateTime::parse_from_str(&date, "%Y-%m-%dT%H:%M:%SZ")?;
            let last_commit_utc = DateTime::<Utc>::from_utc(last_commit, Utc);
            let duration = Local::today().signed_duration_since(last_commit_utc.date());

            if duration.num_days() > 365 {
                entry.deprecated = Some(true);
            } else {
                entry.deprecated = None;
            }
        }
```

Reply:
Yeah that's actually quite nice!